docs.rs failed to build 0.9.0

See https://docs.rs/crate/systemd/0.9.0/builds/395822

```
[INFO] [stderr]  Documenting systemd v0.9.0 (/opt/rustwide/workdir)
[INFO] [stderr] error[E0658]: `#[doc(cfg)]` is experimental
[INFO] [stderr]    --> src/journal.rs:427:24
[INFO] [stderr]     |
[INFO] [stderr] 427 |     #[cfg_attr(docsrs, doc(cfg(feature = "systemd_v245")))]
[INFO] [stderr]     |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[INFO] [stderr]     |
[INFO] [stderr]     = note: see issue #43781 <https://github.com/rust-lang/rust/issues/43781> for more information
[INFO] [stderr]     = help: add `#![feature(doc_cfg)]` to the crate attributes to enable
```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -59,35 +59,18 @@ jobs:
       - name: Install libsystemd-dev
         run: sudo apt-get -o Acquire::Retries=3 install libsystemd-dev
 
-      - name: Check with no-default-features
+      - name: Install cargo-hack
         uses: actions-rs/cargo@v1
         with:
-          command: check
-          args: --all --no-default-features
+          command: install
+          args: cargo-hack
 
-      - name: Check with only bus
+      - name: Check with all variants
         uses: actions-rs/cargo@v1
         with:
-          command: check
-          args: --all --no-default-features --features bus
+          command: hack
+          args: --feature-powerset --skip unstable-doc-cfg check --workspace --all-targets
 
-      - name: Check with only journal
-        uses: actions-rs/cargo@v1
-        with:
-          command: check
-          args: --all --no-default-features --features journal
-
-      - name: Check with only systemd_v245
-        uses: actions-rs/cargo@v1
-        with:
-          command: check
-          args: --all --no-default-features --features systemd_v245
-
-      - name: Check with all features
-        uses: actions-rs/cargo@v1
-        with:
-          command: check
-          args: --all --all-features
   test:
     runs-on: ubuntu-20.04
     strategy:
@@ -115,14 +98,36 @@ jobs:
       - name: Install libsystemd-dev
         run: sudo apt-get -o Acquire::Retries=3 install libsystemd-dev
 
-      - name: Build everything
+      - name: Install cargo-hack
         uses: actions-rs/cargo@v1
         with:
-          command: build
-          args: --all --all-targets
+          command: install
+          args: cargo-hack
+
+      - name: Build everything
+        uses: actions-rs/cargo@v1
+        if: ${{matrix.rust != 'nightly' }}
+        with:
+          command: hack
+          args: --feature-powerset --skip unstable-doc-cfg build --all --all-targets
 
       - name: Run all tests
         uses: actions-rs/cargo@v1
+        if: ${{matrix.rust != 'nightly' }}
         with:
-          command: test
-          args: --all
+          command: hack
+          args: --feature-powerset --skip unstable-doc-cfg test --all
+
+      - name: Build everything
+        uses: actions-rs/cargo@v1
+        if: ${{matrix.rust == 'nightly' }}
+        with:
+          command: hack
+          args: --feature-powerset build --all --all-targets
+
+      - name: Run all tests
+        uses: actions-rs/cargo@v1
+        if: ${{matrix.rust == 'nightly' }}
+        with:
+          command: hack
+          args: --feature-powerset test --all

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -16,6 +16,7 @@ default = ["bus", "journal"]
 bus = ["libsystemd-sys/bus"]
 journal = ["libsystemd-sys/journal"]
 systemd_v245 = ["libsystemd-sys/systemd_v245"]
+unstable-doc-cfg = []
 
 [dependencies]
 log = "~0.4"
@@ -41,7 +42,6 @@ debug = true
 
 [package.metadata.docs.rs]
 all-features = true
-rustdoc-args = ["--cfg", "docsrs"]
 
 [workspace]
 members = ["libsystemd-sys"]

--- a/libsystemd-sys/tests/journal-send.rs
+++ b/libsystemd-sys/tests/journal-send.rs
@@ -1,7 +1,7 @@
-use libsystemd_sys as sd;
-
+#[cfg(feature = "journal")]
 #[test]
 fn raw_send() {
+    use libsystemd_sys as sd;
     let a = ["MESSAGE=raw rust-systemd send"];
     let v = [sd::const_iovec {
         iov_base: a[0].as_ptr() as *const _,

--- a/src/journal.rs
+++ b/src/journal.rs
@@ -424,7 +424,7 @@ impl OpenOptions {
     ///
     /// `sd_journal_open_namespace()`: https://www.freedesktop.org/software/systemd/man/sd_journal_open.html
     #[cfg(feature = "systemd_v245")]
-    #[cfg_attr(docsrs, doc(cfg(feature = "systemd_v245")))]
+    #[cfg_attr(feature = "unstable-doc-cfg", doc(cfg(feature = "systemd_v245")))]
     pub fn open_namespace<A: CStrArgument>(&self, namespace: A) -> Result<Journal> {
         Journal::open_with_opts_ns(Some(namespace), self)
     }

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -1,3 +1,4 @@
+#![cfg_attr(feature = "unstable-doc-cfg", feature(doc_cfg))]
 #![warn(rust_2018_idioms)]
 
 extern crate libsystemd_sys as ffi;

